src/mapInfoContributorManager.cpp

```diff
@@ -0,0 +1,188 @@
+/***************************************************************************
+ *   Copyright (C) 2021 by Piotr Wilczynski - delwing@gmail.com            *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "mapInfoContributorManager.h"
+#include "TArea.h"
+#include "TRoomDB.h"
+
+MapInfoContributorManager::MapInfoContributorManager(QObject* parent, Host* pH) : QObject(parent), mpHost(pH)
+{
+    registerContributor(QStringLiteral("Short"), [=](int roomID, int selectionSize, int areaId, bool showingCurrentArea, QColor& infoColor) {
+        return shortInfo(roomID, selectionSize, areaId, showingCurrentArea, infoColor);
+    });
+    registerContributor(QStringLiteral("Full"), [=](int roomID, int selectionSize, int areaId, bool showingCurrentArea, QColor& infoColor) {
+        return fullInfo(roomID, selectionSize, areaId, showingCurrentArea, infoColor);
+    });
+}
+
+MapInfoContributorManager::~MapInfoContributorManager() {}
+
+void MapInfoContributorManager::registerContributor(QString name, MapInfoCallback callback)
+{   
+    if (contributors.contains(name)) {
+        removeContributor(name);
+    }
+    ordering.append(name);
+    contributors.insert(name, callback);
+}
+
+void MapInfoContributorManager::removeContributor(QString name)
+{
+    mpHost->mMapInfoContributors.remove(name);
+    ordering.removeOne(name);
+    contributors.remove(name);
+}
+
+MapInfoCallback MapInfoContributorManager::getContributor(QString name)
+{
+    return contributors.value(name);
+}
+
+QList<QString> &MapInfoContributorManager::getContributorKeys()
+{
+    return ordering;
+}
+
+MapInfoProperties MapInfoContributorManager::shortInfo(int roomID, int selectionSize, int areaId, bool showingCurrentArea, QColor& infoColor)
```

Comment:
You should put a `Q_UNUSED(`...`);` wrapper line in for each argument that is not used from the function arguments to suppress the compiler warning about it being unused...!

Also having `areaId` as an argument is confusing given that you also have a local automatic variable with the same name - which will mask this argument in the code block that contains the local variable.